src/samplers.jl

```diff
@@ -0,0 +1,151 @@
+
+
+function sampleSMC!(pc::ParticleContainer,resampler::Function =resample_systematic ,resampler_threshold::AbstractFloat = 0.5)
```

Comment:
As mentioned in https://github.com/TuringLang/Turing.jl/pull/990, I think one should not implement the resampling based on ESS explicitly. It feels more natural to me to instead implement a resampler that allows to combine basically any resample algorithm with the ESS checks, as I tried to do in https://github.com/TuringLang/Turing.jl/pull/990. The changes in that PR even allow to reuse the weights computed for evaluation the ESS in the resampling step.

Reply:
Yes, good point! The focus of this pull is mainly on separating the particle filter form Turing. Your changes are not yet included and should definitely be included.

Reply:
Could we not make this step in a second PR?

Reply:
Sure